packages/components/src/Datalist/Datalist.component.test.js

```diff
@@ -214,10 +214,39 @@ describe('Datalist component', () => {
 		input.simulate('blur');
 
 		// then
-		expect(onChange).not.toBeCalled();
+		expect(onChange).not.toHaveBeenCalled();
 		expect(wrapper.find(Typeahead).props().items).toBe(null);
 	});
 
+	it('should change the value on blur in restricted mode and value matches with one suggestion', () => {
+		// given
+		const onChange = jest.fn();
+		const wrapper = mount(
+			<Datalist
+				id="my-datalist"
+				isValid
+				multiSection={false}
+				errorMessage={'This should be correct'}
+				onChange={onChange}
+				{...props}
+				value={'foo'}
+				restricted
+			/>,
+		);
+		const input = wrapper.find('input').at(0);
+		input.simulate('change', { target: { value: 'foobar' } });
```

Comment:
Can you change you use case to have title != value please ?
So we know that in you case, with `{title: 'lol', value: 'mdr'}`, typing `lol` propagates `mdr`